skrifa/src/glyph_name.rs

```diff
@@ -0,0 +1,394 @@
+    len: u8,
+    is_synthesized: bool,
+}
+
+impl GlyphName {
+    /// Returns the underlying name as a string.
+    pub fn as_str(&self) -> &str {
+        let bytes = &self.name[..self.len as usize];
+        core::str::from_utf8(bytes).unwrap_or_default()
+    }
+
+    /// Returns true if the glyph name was synthesized, i.e. not found in any
+    /// source.
+    pub fn is_synthesized(&self) -> bool {
+        self.is_synthesized
+    }
+
+    fn from_bytes(bytes: &[u8]) -> Self {
+        let mut name = Self::default();
+        name.append(bytes);
+        name
+    }
+
+    fn from_post(post: &Post, glyph_id: GlyphId) -> Option<Self> {
+        glyph_id
+            .try_into()
+            .ok()
+            .and_then(|id| post.glyph_name(id))
+            .map(|s| s.as_bytes())
+            .map(Self::from_bytes)
+    }
+
+    fn from_cff_sid(cff: &Cff, sid: Sid) -> Option<Self> {
+        cff.string(sid)
+            .and_then(|s| core::str::from_utf8(s.bytes()).ok())
+            .map(|s| s.as_bytes())
+            .map(Self::from_bytes)
+    }
+
+    fn synthesize(glyph_id: GlyphId) -> Self {
+        use core::fmt::Write;
+        let mut name = Self {
+            is_synthesized: true,
+            ..Self::default()
+        };
+        let _ = write!(GlyphNameWrite(&mut name), "gid{}", glyph_id.to_u32());
+        name
+    }
+
+    fn append(&mut self, bytes: &[u8]) {
+        // We simply truncate when length exceeds the max since glyph names
+        // are expected to be <= 63 chars
+        let start = self.len as usize;
+        let available = MAX_GLYPH_NAME_LEN - start;
+        let copy_len = available.min(bytes.len());
+        self.name[start..start + copy_len].copy_from_slice(&bytes[..copy_len]);
+        self.len = (start + copy_len) as u8;
+    }
+}
+
+impl Default for GlyphName {
+    fn default() -> Self {
+        Self {
+            name: [0; MAX_GLYPH_NAME_LEN],
+            len: 0,
+            is_synthesized: false,
+        }
+    }
+}
+
+impl core::fmt::Debug for GlyphName {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("GlyphName")
+            .field("name", &self.as_str())
+            .field("is_synthesized", &self.is_synthesized)
+            .finish()
+    }
+}
+
+impl core::fmt::Display for GlyphName {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{}", self.as_str())
+    }
+}
+
+impl core::ops::Deref for GlyphName {
+    type Target = str;
+
+    fn deref(&self) -> &Self::Target {
+        self.as_str()
+    }
+}
+
+impl PartialEq<&str> for GlyphName {
+    fn eq(&self, other: &&str) -> bool {
+        self.as_str() == *other
+    }
+}
+
+struct GlyphNameWrite<'a>(&'a mut GlyphName);
+
+impl core::fmt::Write for GlyphNameWrite<'_> {
+    fn write_str(&mut self, s: &str) -> core::fmt::Result {
+        self.0.append(s.as_bytes());
+        Ok(())
+    }
+}
+
+#[derive(Clone)]
+enum Inner<'a> {
```

Comment:
totally unnecessary but I would personally declare this up right below the type that uses it, I think it would make it a bit clearer how the root type works.

Reply:
Makes sense! Done.